model/src/ww3_shel.F90

```diff
@@ -2261,10 +2261,16 @@ PROGRAM W3SHEL
             END IF
 #ifdef W3_OASIS
           ELSE 
-            ID_OASIS_TIME = NINT(DSEC21 ( TIME00 , TIME ))
-            IF ( (DTOUT(7).NE.0) .AND.                               &
-                 (MOD(ID_OASIS_TIME, NINT(DTOUT(7))) .EQ. 0 ) .AND. &
-                 (DSEC21 (TIME, TIMEEND) .GT. 0.0)) DTTST=0.
+            IF ( DTOUT(7).NE.0 ) THEN
+              IF(NINT(DSEC21(TIME00,TIME)) == 0) THEN
+                ID_OASIS_TIME = 0
+                DTTST=0.
+              ELSE
+                ID_OASIS_TIME = NINT(DSEC21 ( TIME00 , TFN(:,J) ))
+                IF ( NINT(MOD(DSEC21(TIME00,TIME), DTOUT(7))) .EQ. 0 .AND. &
+                     DSEC21 (TFN(:,J), TIMEEND) .GT. 0.0 ) DTTST=0.
+              ENDIF
+            ENDIF
```

Comment:
could you add some comment before the IF and ELSE to explain why this splitting of tests

Reply:
@ukmo-juan-castillo could you provide an answer to @mickaelaccensi, I am running regtest at our end in the meantime.